File: 70-climbing-stairs/70-climbing-stairs.py
class Solution:
    global dp
    dp = {}
    def climbStairs(self, n: int) -> int:
        global dp
        if n == 1 or n == 2:
            return n
        if n not in dp:
            dp[n] = self.climbStairs(n-1) + self.climbStairs(n-2)
        return dp[n]

File: 70-climbing-stairs/70-climbing-stairs.java
class Solution {
    int[] dp = new int[46];
    public int climbStairs(int n) {
        if(n == 1 || n == 2) 
            return n;
        if(dp[n] == 0)
            dp[n] = climbStairs(n-1) + climbStairs(n-2);
        return dp[n];
    }
}